docker/docker_client.go

```diff
@@ -258,6 +271,41 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool,
 	}, nil
 }
 
+// CheckAuth validates credentials
+func CheckAuth(ctx context.Context, sCtx *types.SystemContext, username, password, registry string) error {
+	newLoginClient, err := newDockerClient(sCtx, dockerReference{}, false, username, password, registry, "login")
+	if err != nil {
+		return errors.Wrapf(err, "error creating new docker client")
+	}
+
+	resp, err := newLoginClient.makeRequest(ctx, "GET", "/v2/", nil, nil)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return config.SetAuthentication(username, password, newLoginClient.registry)
+	}
+	fmt.Println(resp.Header, resp.Status)
+	return errors.Errorf("error occured with status code %q", resp.StatusCode)
+}
+
+// GetAuth returns the auth if found in auth.json or config.json
+func GetAuth(ctx *types.SystemContext, registry string) (string, string, error) {
+	if registry == dockerHostname {
+		registry = dockerRegistry
+	}
+	return config.GetAuthentication(ctx, registry)
+}
+
+// DeleteAuth removes a registry and its authentication from auth.json
+func DeleteAuth(registry string) (string, error) {
+	if registry == dockerHostname {
+		registry = dockerRegistry
+	}
+	return config.RemoveAuthentication(registry)
```

Comment:
The `registry == dockerHostname` special cases could be directly in the `docker/config` subpackage; that would ensure that they are always processed consistently, and these wrappers could then be removed.

Reply:
fixed